cdc/sink/codec/interface.go

```diff
@@ -52,23 +53,47 @@ type EventBatchEncoder interface {
 
 // MQMessage represents an MQ message to the mqSink
 type MQMessage struct {
-	Key   []byte
-	Value []byte
-	Ts    uint64 // reserved for possible output sorting
+	Key      []byte
+	Value    []byte
+	Ts       uint64              // reserved for possible output sorting
+	Schema   *string             // schema
+	Table    *string             // table
+	Type     model.MqMessageType // type
+	Protocol Protocol            // protocol
 }
 
 // Length returns the expected size of the Kafka message
 func (m *MQMessage) Length() int {
 	return len(m.Key) + len(m.Value)
 }
 
+const physicalShiftBits = 18
+
+// PhysicalTime returns physical time part of Ts in time.Time
+func (m *MQMessage) PhysicalTime() time.Time {
+	ms := int64(m.Ts >> physicalShiftBits)
+	return time.Unix(ms/1e3, (ms%1e3)*1e6)
```

Comment:
This can be simplified with `oracle.GetTimeFromTS(m.Ts)`

Reply:
Sure, It sounds reasonable to reuse code from TiDB.

Reply:
Fixed